### PR TITLE
arrow: Update to 16.0.0

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

**Comment by raulcd** [comment]
it seems the patches have to be updated, I'll work on fixing this.

**Comment by Biswa96** [comment]
Removing the 001-support-llvm-18.patch in arrow should fix the build.

File: mingw-w64-arrow/PKGBUILD [hunk 1]
```diff
@@ -49,7 +49,7 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
 source=("https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz"
         "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz.asc"
         "001-support-llvm-18.patch::https://github.com/apache/arrow/commit/ed78986a.patch")
```
Comment (kou) [inline mingw-w64-arrow/PKGBUILD:51]: Could you also remove this line?

File: mingw-w64-arrow/PKGBUILD [hunk 2]
```diff
@@ -49,7 +49,7 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
...
-sha256sums=('abbf97176db6a9e8186fe005e93320dac27c64562755c77de50a882eb6179ac6'
+sha256sums=('9f4051ae9473c97991d9af801e2f94ae3455067719ca7f90b8137f9e9a700b8d'
             'SKIP'
             'f7516c9e5b929b4b7ae3d8da1f3ec2ad5fffdc5bf229af5e00ae4fad2ecfcd07')
```
Comment (kou) [inline mingw-w64-arrow/PKGBUILD:54]: This line too.

**Comment by raulcd** [comment]
@Biswa96 this has been fixed now, CI is passing. I just rebased to have a clean history. Thanks!

File: mingw-w64-arrow/PKGBUILD [hunk 3]
```diff
@@ -48,10 +48,10 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-xsimd")
 source=("https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz"
         "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${pkgver}/apache-arrow-${pkgver}.tar.gz.asc"
-        "001-support-llvm-18.patch::https://github.com/apache/arrow/commit/ed78986a.patch")
-sha256sums=('abbf97176db6a9e8186fe005e93320dac27c64562755c77de50a882eb6179ac6'
+        "0001-find-protobuf-config-instead-of-module.patch")
```
Comment (kou) [inline mingw-w64-arrow/PKGBUILD:51]: Could you use the merged change instead? (We also need to update `sha256sum`.)

```suggestion
        "0001-find-protobuf-config-instead-of-module.patch::https://github.com/apache/arrow/commit/64be7a2.patch")
```

Comment (raulcd) [reply mingw-w64-arrow/PKGBUILD]: I've updated the patch and the sha256 but I don't seem to know how to calculate the correct sha for a patch download like this. I've tried with this:
```
$ curl https://github.com/apache/arrow/commit/64be7a2.patch -o 0001-find-protobuf-config-instead-of-module.patch
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  2511  100  2511    0     0  13947      0 --:--:-- --:--:-- --:--:-- 14027
$ sha256sum 0001-find-protobuf-config-instead-of-module.patch 
2eb48799459ea196bf66e0578a97bd7f0ccc1072f6b48c8b260f80f8740ec11f  0001-find-protobuf-config-instead-of-module.patch
```
but this doesn't seem to be the correct sha. Can someone help me understand how I can check the sha for this?

File: mingw-w64-arrow/0001-find-protobuf-config-instead-of-module.patch
Comment (kou) [inline mingw-w64-arrow/0001-find-protobuf-config-instead-of-module.patch:1]: Could you remove this file?
If we have this file, https://github.com/apache/arrow/commit/64be7a2.patch isn't downloaded.

Comment (raulcd) [reply mingw-w64-arrow/0001-find-protobuf-config-instead-of-module.patch:1]: oops, I forgot to remove the patch. Thanks @kou

**Pull request merged**